File: youtube_logic.py
import os
import requests

def get_latest_videos(channel_id: str, api_key: str | None = None, max_results: int = 5):
    """Return a list of the latest videos from a YouTube channel."""
    api_key = api_key or os.getenv("YOUTUBE_API_KEY")
    if not api_key:
        raise ValueError("YOUTUBE_API_KEY is not set")
    url = "https://www.googleapis.com/youtube/v3/search"
    params = {
        "key": api_key,
        "channelId": channel_id,
        "part": "snippet",
        "order": "date",
        "type": "video",
        "maxResults": max_results,
    }
    resp = requests.get(url, params=params, timeout=10)
    resp.raise_for_status()
    items = resp.json().get("items", [])
    videos = [
        {
            "id": item.get("id", {}).get("videoId"),
            "title": item.get("snippet", {}).get("title"),
        }
        for item in items
        if "videoId" in item.get("id", {})
    ]
    return {"videos": videos}

File: youtube_logic.js
async function getLatestVideos(channelId, apiKey = process.env.YOUTUBE_API_KEY, maxResults = 5) {
  if (!apiKey) {
    throw new Error('YOUTUBE_API_KEY is not set');
  }
  const url = 'https://www.googleapis.com/youtube/v3/search';
  const params = new URLSearchParams({
    key: apiKey,
    channelId,
    part: 'snippet',
    order: 'date',
    type: 'video',
    maxResults: String(maxResults)
  });
  const res = await fetch(`${url}?${params.toString()}`);
  if (!res.ok) {
    throw new Error(`YouTube API error ${res.status}`);
  }
  const data = await res.json();
  const videos = (data.items || [])
    .filter(it => it.id && it.id.videoId)
    .map(it => ({ id: it.id.videoId, title: it.snippet?.title }));
  return { videos };
}

module.exports = { getLatestVideos };
